mu-plugins/central-hub/tests/phpunit/unit/config-store/getAllKeys.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ *  Tests for getAllKeys()
+ *
+ * @package    spiralWebDb\centralHub\Tests\Unit\ConfigStore
+ * @since      1.3.0
+ * @author     Robert A. Gadon
+ * @link       https://github.com/rgadon107/cornerstone
+ * @license    GNU General Public License 2.0+
+ */
+
+namespace spiralWebDb\centralHub\Tests\Unit\ConfigStore;
+
+use Brain\Monkey;
+use function KnowTheCode\ConfigStore\getAllKeys;
+use spiralWebDb\Cornerstone\Tests\Unit\Test_Case;
+
+/**
+ * Class Tests_GetAllKeys
+ *
+ * @package spiralWebDb\centralHub\Tests\Unit\ConfigStore
+ * @group   config-store
+ */
+class Tests_GetAllKeys extends Test_Case {
+
+	/**
+	 * Prepares the test environment before each test.
+	 */
+	protected function setUp() {
+		parent::setUp();
+
+		require_once CENTRAL_HUB_ROOT_DIR . '/src/config-store/api.php';
+	}
+
+	/**
+	 * Test getAllKeys() should return the array keys from `_the_store()`.
+	 */
+	public function test_should_return_the_array_keys_from_the_store() {
+		Monkey\Functions\expect( 'array_keys' )
+			->once()
+			->with( 'KnowTheCode\ConfigStore\_the_store' )
+			->andReturnValues( [ 'config_store[store_key]' ] );
+		getAllKeys();
```

Comment:
@rgadon107 Here you want to test what you get back to ensure you get an array of keys.

Reply:
@rgadon107 Also, with changes to `_the_store()` will you get all of the contents back when not sending a store key? This function is requesting ALL OF THE CONTENTS of the store.  It then extracts all of the store keys.  But the `_the_store` function has changed.  Right?  

So how can you request the entire store?  Think about it.   One way is to send some value as the store key that means "Hey, give me everything in the store."

Reply:
@hellofromtonya I can't validate that the API function `getAllKeys()` is indeed getting all the keys from `_the_store`. I keep getting the error message:

>spiralWebDb\centralHub\Tests\Unit\ConfigStore\Tests_GetAllKeys::test_should_return_the_array_keys_from_the_store

>array_keys() expects parameter 1 to be array, null given

When I use `var_dump()` on the `_the_store( 'foo', $config )`, I can see that the store is storing a valid key and config. But that store key is not getting passed on to the API function`getAllKeys` when I run the assert on it. 

Did I make a mistake in how I set up the expectation for `_the_store` with BrainMonkey?

Reply:
Think about what this function does.  It wants ALL of the store keys that are currently stored in the store. How can you get ALL keys from the store?

Let's think about what should happen:

1. `getAllKeys()` requests all of the data from `_the_store()`: "Hey store, gimme all your data."
2. `_the_store()`then returns its static variable `return $config_store`.

Let's look at the real function to find out what we need to provide to get all of the store keys:

```php
function _the_store( $store_key = '', $config_to_store = array(), $remove = null ) {
	static $config_store = array();

	// Bail out if no store key is given.
	if ( ! $store_key ) {
		return $config_store;
	}
	// rest of the code left out for brevity.
}
```

To get everything, all you do is invoke `_the_store()` without a store key.  Right?  Look at the above code.

Then in your unit test, you need to mock that behavior. 

1. No arguments will be received, i.e. _with_
2. It should return everything stored in it.

Let's modify the mock.  I'll go step-by-step to help you visualize what's happening.

```php
->with( 'foo', $config )
```

There will be no arguments.  We can change this to `withNoArgs()`.

```php
->andReturnValues( [ 'foo' ] );
```

The store will return whatever is stored in it.  Everything.  So you want to create a structure that emulates how it will work in production, i.e.:

store key => config parameters.

I updated the code here https://github.com/rgadon107/cornerstone/pull/24/commits/fe4ee2143b2db7cb7e26bc0715fc9b54e5ff7bbe.  Take a look and ask any questions.

Reply:
@hellofromtonya I understand the changes you demonstrated. I've pulled them into my local repo on the `add/getAllKeys` branch.